Scripts/Runtime/Util/Random/RandomUtil.cs

```diff
@@ -0,0 +1,45 @@
+using System.Runtime.CompilerServices;
+using Unity.Mathematics;
+using DateTime = System.DateTime;
+using RuntimeInitializeLoadType = UnityEngine.RuntimeInitializeLoadType;
+
+namespace Anvil.Unity.DOTS.Util
+{
+    /// <summary>
+    /// Utility class for simplifying getting access to a <see cref="Random"/> instance.
+    /// </summary>
+    public static class RandomUtil
+    {
+        private static Random s_InternalRandom;
+
+        //With DOTS, it's recommended to disable the Domain Reload when entering play mode as it will be pretty slow.
+        //https://docs.unity3d.com/Packages/com.unity.entities@0.17/manual/install_setup.html
+        //Because of this, statics will preserve state across play sessions. By using the RuntimeInitializeOnLoadMethod
+        //we can ensure we reset the state on every play session.
+        //RuntimeInitializeLoadType.SubsystemRegistration represents the earliest moment of all the entries in the enum.
+        [UnityEngine.RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        private static void Init()
+        {
+            s_InternalRandom = new Random((uint)DateTime.Now.Ticks);
+        }
+        
+        /// <summary>
+        /// Creates a new instance of a <see cref="Random"/> and gives it a random seed between 1 and uint.MaxValue
+        /// </summary>
+        /// <remarks>
+        /// While you could create a new instance yourself, choosing the seed can be annoying. Common approaches are to
+        /// use a different random number generator to generate the seed value or use some aspect of the current time.
+        /// The issue with using the current time is that if you create a bunch of <see cref="Random"/> instance at the
+        /// same time in a loop, the seeds will all be the same and the random numbers generated will all be the same.
+        /// This method uses an internal instance of <see cref="Random"/> to generate a new seed and ensures that it
+        /// fits the range requirements of the seed not being 0. You are guaranteed to have different seeds each time
+        /// this is called.
```

Comment:
```suggestion
        /// fits the range requirements of the seed not being 0.
```

That's not a guarantee. Random can generate the same value multiple times in a row.

Reply:
I've deleted that line because you are technically correct. 

However, because it's a PRNG, I'm not sure that `NextUInt` could return the same value multiple times in a row since we're using the whole range of 1 to uint.MaxValue. 

If we were doing `NextUInt` from 1 to 2 then sure, we'd collapse down to multiple 1's or 2's in a row. But having the full range and getting the same value multiple times in a row would just have us be in an endless loop no? The state generated that gets fed in returns the same state which therefore gets fed back in and on and on we go.

Reply:
We talked about it and we agree. All good here.